UnityProject/Assets/Scripts/Weapons/Gun.cs

```diff
@@ -0,0 +1,889 @@
+		}
+
+		#endregion
+
+
+		#region Weapon Firing Mechanism
+
+		private void UpdateMe()
+		{
+			//don't process if we are server and the gun is not held by anyone
+			if (isServer && serverHolder == null) return;
+
+			//if we are client, make sure we've initialized
+			if (!isServer && !PlayerManager.LocalPlayer) return;
+
+			//if we are client, only process this if we are holding it
+			if (!isServer)
+			{
+				if (UIManager.Hands == null || UIManager.Hands.CurrentSlot == null) return;
+				var heldItem = UIManager.Hands.CurrentSlot.ItemObject;
+				if (gameObject != heldItem) return;
+			}
+
+			//update the time until the next shot can happen
+			if (FireCountDown > 0)
+			{
+				FireCountDown -= Time.deltaTime;
+				//prevents the next projectile taking miliseconds longer than it should
+				if (FireCountDown < 0)
+				{
+					FireCountDown = 0;
+				}
+			}
+
+			//remaining logic is server side only
+
+			//this will only be executed on the server since only the server
+			//maintains the queued actions
+			if (queuedShots.Count > 0 && FireCountDown <= 0)
+			{
+				//fire the next shot in the queue
+				DequeueAndProcessServerShot();
+			}
+
+			if (queuedUnload && queuedShots.Count == 0 && allowMagazineRemoval && !MagInternal)
+			{
+				// done processing shot queue,
+				// perform the queued unload action, causing all clients and server to update their version of this Weapon
+				// due to the syncvar hook
+				// there should not be an unload action for internal magazines
+				Inventory.ServerDrop(magSlot);
+				queuedUnload = false;
+
+			}
+			if (queuedLoadMagNetID != NetId.Invalid && queuedShots.Count == 0)
+			{
+				//done processing shot queue, perform the reload, causing all clients and server to update their version of this Weapon
+				//due to the syncvar hook
+				if (MagInternal)
+				{
+					var clip = NetworkIdentity.spawned[queuedLoadMagNetID];
+					MagazineBehaviour clipComp = clip.GetComponent<MagazineBehaviour>();
+					string message = CurrentMagazine.LoadFromClip(clipComp);
+					Chat.AddExamineMsg(serverHolder, message);
+					queuedLoadMagNetID = NetId.Invalid;
+				}
+				else
+				{
+					var magazine = NetworkIdentity.spawned[queuedLoadMagNetID];
+					var fromSlot = magazine.GetComponent<Pickupable>().ItemSlot;
+					Inventory.ServerTransfer(fromSlot, magSlot);
+					queuedLoadMagNetID = NetId.Invalid;
+				}
+			}
+		}
+
+		/// <summary>
+		/// attempt to reload the weapon with the item given
+		/// </summary>
+		private void TryReload(GameObject ammo)
+		{
+			MagazineBehaviour magazine = ammo.GetComponent<MagazineBehaviour>();
+			if (CurrentMagazine == null || (MagInternal && magazine.isClip))
+			{
+				//RELOAD
+				// If the item used on the gun is a magazine, check type and reload
+				AmmoType ammoType = magazine.ammoType;
```

Comment:
Codacy found an issue: [Rename 'ammoType' which hides the field with the same name.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)